internal/pkg/container/kernel.go

```diff
@@ -0,0 +1,47 @@
+package container
+
+import (
+	"path"
+	"path/filepath"
+	"sort"
+
+	"github.com/hpcng/warewulf/internal/pkg/wwlog"
+)
+
+var (
+	kernelSearchPaths = []string{
+		// This is a printf format where the %s will be the kernel version
+		"/boot/vmlinuz-*",
+		"/lib/modules/*/vmlinuz*",
+	}
+)
+
+func KernelFind(container string) string {
+	container_path := RootFsDir(container)
+	if container_path == "" {
+		return ""
+	}
+
+	for _, searchPath := range kernelSearchPaths {
+
+		check_path := path.Join(container_path, searchPath)
+
+		wwlog.Printf(wwlog.DEBUG, "Searching for kernel(s) at: %s\n", check_path)
+		kernels, err := filepath.Glob(check_path)
+		if err != nil {
+			return ""
+		}
+
+		if len(kernels) > 1 {
+			sort.Slice(kernels, func(i, j int) bool {
+				return kernels[i] > kernels[j]
+			})
+			wwlog.Printf(wwlog.VERBOSE, "Multiple kernels found in container: %s\n", container)
```

Comment:
Consider if there is an API around this. The caller won't see the log message.

Reply:
Yep, but that's gonna be a much bigger change I think, unless you are suggesting we return verbose and/or `error` messages to the caller?

Reply:
For now it may be best to just log. Plan is for the API to call identical code as the CLI.

Reply:
This will have to be a bigger fix for API integration. For now, I'm going to leave this as it stands for compatibility with existing standard (all of which we will go back and fix at once once we've defined the best way to do it).

Thanks and great thinking!